api/survey.ts

```diff
@@ -0,0 +1,59 @@
+import handler from './handler';
+import db from './db';
+
+export default handler(async (request : any) => {
+
+  let connection = db();
+  const client = await connection.connect()
+
+  // Take data from request and build sql call from it
+  // TODO: Move this data structure to it's own class
+  const {
+    //Sharded Values
+    patientid,
+    completiondate,
+    completiontime,
+    //TFI Survey
+    tfisurveyid,
+    tfi_i,
+    tfi_sc,
+    tfi_c,
+    tfi_si,
+    tfi_a,
+    tfi_r,
+    tfi_q,
+    tfi_e,
+    tfi_overallscore,
+    //THS Survey
+    thssurveyid,
+    ths_sectiona,
+    ths_sectionb,
+    ths_sectionc,
+    //TS Survey
+    tssurveyid,
+    ts_type
+  } = request.body
+
+  var tfisurvey_sql = "INSERT INTO tfisurvey (tfisurveyid, patientid, completiondate, completiontime, tfi_i, tfi_sc, tfi_c, tfi_si, tfi_a, tfi_r, tfi_q, tfi_e, tfi_overallscore) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)"
+  var tfisurvey_values = [tfisurveyid, patientid, completiondate, completiontime, tfi_i, tfi_sc, tfi_c, tfi_si, tfi_a, tfi_r, tfi_q, tfi_e, tfi_overallscore]
```

Comment:
It would be nice to rewrite this function such that each of these strings (`tfi_i`, etc) only appears once per query it appears in. Eg:
`const tfiNames = ["tfisurveyid", "patientid", ...]`
`const tfisurveySql = 'INSERT INTO tfisurvey (' + tfiNames.join(', ') + ') VALUES (' + tfiNames.map((v, i) => '$' + (i + 1) + ')';`
`const tfisurveyValues = tfiNames.map((name) => request.body[name])`

Reply:
Bonus points if you do this in a loop or function so that you re-use this code for all of the survey tables, rather than copy/pasting.